packages/multi-column/src/extensions/ColumnResize/ColumnResizeExtension.ts

```diff
@@ -0,0 +1,376 @@
+    const rightColumnPos = rightColumnNodeAndPos.posBeforeNode;
+
+    return {
+      type: "hover",
+      leftColumn: {
+        element: leftColumnElement,
+        id: leftColumnId,
+        node: leftColumnNode,
+        pos: leftColumnPos,
+      },
+      rightColumn: {
+        element: rightColumnElement,
+        id: rightColumnId,
+        node: rightColumnNode,
+        pos: rightColumnPos,
+      },
+    };
+  };
+
+  // When the user mouses down near the boundary between two columns, we
+  // want to set the plugin state to resize, so the columns can be resized
+  // by moving the mouse.
+  mouseDownHandler = (event: MouseEvent) => {
+    let newState: ColumnState = this.getColumnHoverOrDefaultState(event);
+    if (newState.type === "default") {
+      return;
+    }
+
+    event.preventDefault();
+
+    const startPos = event.clientX;
+
+    const leftColumnWidthPx =
+      newState.leftColumn.element.getBoundingClientRect().width;
+    const rightColumnWidthPx =
+      newState.rightColumn.element.getBoundingClientRect().width;
+
+    const leftColumnWidthPercent = newState.leftColumn.node.attrs
+      .width as number;
+    const rightColumnWidthPercent = newState.rightColumn.node.attrs
+      .width as number;
+
+    newState = {
+      type: "resize",
+      startPos,
+      leftColumn: {
+        ...newState.leftColumn,
+        widthPx: leftColumnWidthPx,
+        widthPercent: leftColumnWidthPercent,
+      },
+      rightColumn: {
+        ...newState.rightColumn,
+        widthPx: rightColumnWidthPx,
+        widthPercent: rightColumnWidthPercent,
+      },
+    };
+
+    console.log(newState);
+    this.view.dispatch(
+      this.view.state.tr.setMeta(columnResizePluginKey, newState)
+    );
+  };
+
+  // If the plugin isn't in a resize state, we want to update it to either a
+  // hover state if the mouse is near the boundary between two columns, or
+  // default otherwise. If the plugin is in a resize state, we want to
+  // update the column widths based on the horizontal mouse movement.
+  mouseMoveHandler = (event: MouseEvent) => {
+    const pluginState = columnResizePluginKey.getState(this.view.state);
+    if (!pluginState) {
+      return;
+    }
+
+    // If the user isn't currently resizing columns, we want to update the
+    // plugin state to maybe show or hide the resize border between columns.
+    if (pluginState.type !== "resize") {
+      const newState = this.getColumnHoverOrDefaultState(event);
+
+      // Prevent unnecessary state updates (when the state before and after
+      // is the same).
+      const bothDefaultStates =
+        pluginState.type === "default" && newState.type === "default";
+      const sameColumnIds =
+        pluginState.type !== "default" &&
+        newState.type !== "default" &&
+        pluginState.leftColumn.id === newState.leftColumn.id &&
+        pluginState.rightColumn.id === newState.rightColumn.id;
+      if (bothDefaultStates || sameColumnIds) {
```

Comment:
why ||?

Reply:
`bothDefaultStates` and `sameColumnIds` check 2 separate cases:

1. The column drag bar was not visible before and is still not visible.
2. The column drag bar was visible before and is still visible, but between the same pair of columns.

It wouldn't make sense to use `&&` here, as in one case we check if both states have type `"default"`, and in the other we check if neither of them have it.